### PR TITLE
upgrade(remoteConfig): Add top-level enablement parameter

### PR DESCRIPTION
**Pull request by BaptisteFoy**
#### What this PR does / why we need it:
Adds a top-level parameter to enable Remote Configuration both on the datadog agent and on the cluster agent. This top-level parameter will override the two existing parameters that enable Remote Configuration on the two agents separately.

This will avoid confusion in the scenarii where Remote Configuration is enabled for one but not the other.

#### Checklist
- [x] Chart Version bumped
- [x] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [x] `CHANGELOG.md` has been updated
- [x] Variables are documented in the `README.md`

### CODE REVIEW

File: charts/datadog/templates/_containers-common-env.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@
     secretKeyRef:
       name: {{ template "datadog.apiSecretName" . }}
       key: api-key
-{{- if .Values.datadog.remoteConfiguration.enabled }}
+{{- if or (.Values.datadog.remoteConfiguration.enabled .Values.remoteConfiguration.enabled) }}
```
Comment (mellon85) [inline charts/datadog/templates/_containers-common-env.yaml:12]: can't we set a variable to all these conditions with an `or` and use that variable?
One for the controller, and one for the agent. So we have a common global definition to use in these places

Comment (BaptisteFoy) [reply charts/datadog/templates/_containers-common-env.yaml]: Not a big fan of what this makes us do with the string comparison, but done in [20303ce](https://github.com/DataDog/helm-charts/pull/1058/commits/20303ce608848ebe087e9f6b489f2ed3138384ed).

**Review by mellon85 — APPROVED** [review]

**Review by mellon85 — APPROVED** [review]

File: charts/datadog/Chart.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: datadog
-version: 3.30.6
+version: 3.30.7
```
Comment (AliDatadog) [inline charts/datadog/Chart.yaml:3]: I think we need to update the minor version when a parameter is added

File: charts/datadog/values.yaml [hunk 3]
```diff
@@ -1935,3 +1936,9 @@ providers:
   aks:
     # providers.aks.enabled -- Activate all specifities related to AKS configuration. Required as currently we cannot auto-detect AKS.
     enabled: false
+
+remoteConfiguration:
+  # remoteConfiguration.enabled -- Set to true to enable remote configuration on the cluster agent (if set) and the datadog agent.
```
Comment (celenechang) [inline charts/datadog/values.yaml:1941]:
```suggestion
  # remoteConfiguration.enabled -- Set to true to enable remote configuration on the Cluster Agent (if set) and the node agent.
```

File: charts/datadog/Chart.yaml (hunk 2 above)
Comment (BaptisteFoy) [reply charts/datadog/Chart.yaml]: Fixed in [e9615a0](https://github.com/DataDog/helm-charts/pull/1058/commits/e9615a0afc93846fa187050ec5c9ff4e837885e5)

File: charts/datadog/values.yaml (hunk 3 above)
Comment (BaptisteFoy) [reply charts/datadog/values.yaml]: Fixed in [bce36c7](https://github.com/DataDog/helm-charts/pull/1058/commits/bce36c7cb465befe6dd59927e6e6265fe91e5112)

**Review by celenechang — CHANGES_REQUESTED** [review]
looks good, do you mind running the helm-docs.sh script again?

**Comment by BaptisteFoy** [comment]
`helm-docs` ran in [77a642d](https://github.com/DataDog/helm-charts/pull/1058/commits/77a642d3e3c939f77d8374ed77038a16140a8f79)!

**Review by celenechang — APPROVED** [review]

**Pull request merged**